include/deal.II/lac/orthogonalization.h

```diff
@@ -0,0 +1,49 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2023 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_lac_orthogonalization_h
+#define dealii_lac_orthogonalization_h
+
+
+#include <deal.II/base/config.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace LinearAlgebra
+{
+  /**
+   * Supporeted orthogonalization strategies within SolverGMRES and
+   * SolverFGMRES.
+   */
+  enum class OrthogonalizationStrategy
+  {
+    /**
+     * Use modified Gram-Schmidt algorithm.
+     */
+    modified_gram_schmidt,
+    /**
+     * Use classical Gram-Schmidt algorithm. Since this approach works on
+     * multi-vectors and performs a global reduction only once, it is
+     * more efficient than the modified Gram-Schmidt algorithm.
+     * However, it might be numerically unstable.
```

Comment:
both variants of Gram-Schmidt are numerically unstable. But the classical algorithm is considerably less stable.
```suggestion
     * However, it is less stable in terms of roundoff error propagation, requiring
     * additional re-orthogonalization steps more frequently.
```

Reply:
This comment makes me think if we do the right thing in the re-orthogonalization for Gram-Schmidt. Intuitively, I think the classical Gram-Schmidt might have slightly different requirements than the modified one. @peterrum do you see immediately if we treat all subsequent vectors correctly with the present code? Otherwise, I suggest we open an issue to look at it when there is more time or a concrete need, because that would mean to look things up in the literature.

Reply:
I have created an issue: https://github.com/dealii/dealii/issues/14979